subprojects/code-quality/src/integTest/groovy/org/gradle/api/plugins/quality/checkstyle/CheckstylePluginVersionIntegrationTest.groovy

```diff
@@ -520,6 +520,8 @@ dependencies {
     implementation localGroovy()
 }
 
+testing.suites.test.useJUnit()
```

Comment:
This effectively adds an `implementation junit:junit:4.13` dependency declaration. Since the tests all call `check`, we need to make sure we have junit at runtime to avoid the deprecation warning.